apps/meteor/ee/tests/unit/apps/livechat-enterprise/lib/AutoCloseOnHold.tests.ts

```diff
@@ -0,0 +1,246 @@
+		info: infoStub,
+		debug: debugStub,
+	}),
+};
+
+const mocks = {
+	'@rocket.chat/agenda': { Agenda: MockAgendaClass },
+	'meteor/meteor': { Meteor: { startup: mockMeteorStartup } },
+	'meteor/mongo': {
+		MongoInternals: {
+			defaultRemoteCollectionDriver: () => {
+				return {
+					mongo: { client: { db: sinon.stub() } },
+				};
+			},
+		},
+	},
+	'../../../../../app/livechat/server/lib/LivechatTyped': { Livechat: { closeRoom: mockLivechatCloseRoom } },
+	'./logger': { schedulerLogger: mockLogger },
+	'@rocket.chat/models': {
+		LivechatRooms: mockLivechatRooms,
+		Users: mockUsers,
+	},
+};
+
+const { AutoCloseOnHoldSchedulerClass } = proxyquire
+	.noCallThru()
+	.load('../../../../../app/livechat-enterprise/server/lib/AutoCloseOnHoldScheduler', mocks);
+
+describe('AutoCloseOnHoldScheduler', () => {
+	beforeEach(() => {
+		mockMeteorStartup.resetHistory();
+	});
+
+	it('should call logger.section upon instantiating', () => {
+		expect(mockLogger.section.called).to.be.true;
+	});
+
+	describe('init', () => {
+		beforeEach(() => {
+			mockAgendaStart.resetHistory();
+			mockAgendaScheduler.resetHistory();
+		});
+
+		it('should do nothing if scheduler is already running', async () => {
+			const scheduler = new AutoCloseOnHoldSchedulerClass();
+			scheduler.running = true;
+
+			await scheduler.init();
+
+			expect(mockAgendaScheduler.called).to.be.false;
+		});
+
+		it('should succesfully init the scheduler', async () => {
+			const scheduler = new AutoCloseOnHoldSchedulerClass();
+
+			await scheduler.init();
+
+			expect(mockAgendaStart.calledOnce).to.be.true;
+			expect(scheduler.running).to.be.true;
+			expect(infoStub.calledWith('Service started')).to.be.true;
+		});
+	});
+
+	describe('scheduleRoom', () => {
+		beforeEach(() => {
+			mockAgendaCancel.resetHistory();
+			mockAgendaDefine.resetHistory();
+		});
+
+		it('should schedule a room', async () => {
+			const scheduler = new AutoCloseOnHoldSchedulerClass();
+
+			await scheduler.init();
+			await scheduler.scheduleRoom('roomId', 5, 'test comment');
+
+			const myScheduleTime = moment(new Date()).add(5, 's').toDate();
+			expect(mockAgendaCancel.calledBefore(mockAgendaDefine)).to.be.true;
+			expect(mockAgendaCancel.calledWith({ name: 'omnichannel_auto_close_on_hold_scheduler-roomId' }));
+			expect(mockAgendaDefine.calledWithMatch('omnichannel_auto_close_on_hold_scheduler-roomId'));
+			const funcScheduleTime = mockAgendaScheduler.getCall(0).firstArg;
+
+			expect(funcScheduleTime).to.be.closeToTime(myScheduleTime, 5);
+			expect(mockAgendaScheduler.calledWithMatch('omnichannel_auto_close_on_hold_scheduler-roomId'));
+		});
+	});
+
+	describe('unscheduleRoom', () => {
+		beforeEach(() => {
+			mockAgendaCancel.resetHistory();
+		});
+
+		it('should fail if scheduler has not been init', async () => {
+			const scheduler = new AutoCloseOnHoldSchedulerClass();
+
+			try {
+				await scheduler.unscheduleRoom('roomId');
+			} catch (e) {
+				expect(e).to.not.to.be.undefined;
```

Comment:
`[].not.be.undefined`

![image](https://github.com/RocketChat/Rocket.Chat/assets/17487063/27a725e8-e539-4602-a6d5-6d026cf85588)

Reply:
```suggestion
				expect(e).not.to.be.undefined;
```